edd-discount-widget.php

```diff
@@ -134,21 +140,61 @@ function form( $instance ) {
         }
 
 
-        function get_discounts( $site_url, $api_key, $api_token, $max_discounts ) {
+        /**
+         * Obtains the EDD discounts from the site URL.
+         * 
+         * @param mixed $site_url      The URL to extract the discounts from.
+         * @param mixed $api_key       The API key for the site.
+         * @param mixed $api_token     The API token for the site.
+         * 
+         * @return array|WP_Error      An array of discounts or a WP_Error object.
+         */
+        function get_discounts( $site_url, $api_key, $api_token ) {
+
+            $site_url  = sanitize_text_field( $site_url );
+            $api_key   = sanitize_text_field( $api_key );
+            $api_token = sanitize_text_field( $api_token );
+
+            if ( empty( $site_url ) || empty( $api_key ) || empty( $api_token ) ) {
+                return new WP_Error(
+                    'edd-discounts-no-settings',
+                    __( 'Widget settings are incomplete.', 'edd-discounts-widget' )
+                );
+            }
+
             $options = array(
                 'timeout'   => 5
             );
 
             $temp_url = parse_url( $site_url );
-            if( !$temp_url['scheme'] == 'http' && !$temp_url['scheme'] == 'https' )
+            if ( isset( $temp_url['scheme'] ) && ! in_array( $temp_url['scheme'], array( 'http', 'https' ), true ) ) {
```

Comment:
I think this may need to be an `empty` check after all--just tested and if I set my URL to `edd.local`, I get "A valid URL was not provided" as the error. Logging the `$temp_url`, I have:

```
(
    [path] => edd.local
)
```

So the URL is not updated and the request fails.

Reply:
Ah so maybe what we actually need is:

```php
if ( empty( $temp_url['scheme'] ) || ( isset( $temp_url['scheme'] ) && ! in_array( $temp_url['scheme'], array( 'http', 'https' ), true ) ) {
```